examples/basic/plot_transform_until.py

```diff
@@ -0,0 +1,89 @@
+"""
+Preprocessing with zscore and PCA
+===================================
+
+This example uses the 'iris' dataset, performs simple binary
+classification after the features are normalized using zscore and feature
+reduction using PCA. We will check the features after each preprocessing
+step.
+
+"""
+
+# Authors: Shammi More <s.more@fz-juelich.de>
+#
+# License: AGPL
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+from seaborn import load_dataset
+
+from julearn import run_cross_validation
+from julearn.utils import configure_logging
+
+###############################################################################
+# Set the logging level to info to see extra information
+configure_logging(level='INFO')
+
+###############################################################################
+# Load the iris data from seaborn
+df_iris = load_dataset('iris')
+
+###############################################################################
+# The dataset has three kind of species. We will keep two to perform a binary
+# classification.
+
+df_iris = df_iris[df_iris['species'].isin(['versicolor', 'virginica'])]
+
+###############################################################################
+# We will use the sepal length, width and petal length and
+# petal width as features and predict the species
+
+X = ['sepal_length', 'sepal_width', 'petal_length', 'petal_width']
+y = 'species'
+
+###############################################################################
+# Let's look at the summary statistics of the raw features
+print('Summary Statistics of the raw features : \n', df_iris.describe())
+
+###############################################################################
+# We will preprocess the features (zscore and PCA) first before we train a
+# random forest model
+
+# Define the model parameters and preprocessing steps first
+model_params = {'pca__n_components': 2, 'rf__n_estimators': 200, 'cv': 5}
+preprocess_X = ['zscore', 'pca']
+
+scores, model = run_cross_validation(
+    X=X, y=y, data=df_iris, model='rf', preprocess_X=preprocess_X,
+    scoring=['accuracy', 'roc_auc'], model_params=model_params,
+    return_estimator='final', seed=200)
+
+###############################################################################
+# We can now look at zscored features
```

Comment:
Please add a bit more explanation what and how preprocess works to also include this information: 

preprocess method can outputs the data after preprocessing. by default with all preprocessing steps, but if you set the agrument until only up top (inclusive) the named preprocessing step. The name of the preprocessing step is the same as used in the run_cross_validation function in preprocess_X.

If its easier to explain, you can also start with an example where you only do preprocess without specifying until.